Fix `payu-dev` tests when loading `backports.zstd._cffi`

There's currently an error when testing the `payu-dev` environments. (See https://github.com/ACCESS-NRI/model-release-condaenv/actions/runs/20043204162/job/57491226445). The tests run pytests that attempt to import all accessible Python modules in the environment.

The exception raised in the test that tries loading `backports.zstd._cffi` is
 `ModuleNotFoundError: No module named 'backports.zstd._zstd_cffi'`

<details>
<summary>Failure test error logs</summary>

```
=================================== FAILURES ===================================
______________________________ test_walk_packages ______________________________

    def test_walk_packages():
        global exception
        config = import_config('../environments/'+ os.environ['CONDA_ENVIRONMENT'] + '/testconfig.yml')
        for mod in config.get('preload',[]):
            try:
                if debug: print("Preloading {}".format(mod))
                __import__(mod)
            except:
                print("Error preloading {}".format(mod))

        exception = set(config.get('exception',[]))
        skip = set(config.get('skip',[]))
        if debug: print('Exception: {}'.format(exception))
        if debug: print('Skip: {}'.format(skip))
>       for importer, name, ispkg in my_walk_packages(onerror=handle_error,skip=skip):

test_environment.py:134:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_environment.py:83: in my_walk_packages
    yield from my_walk_packages(path, info.name+'.', onerror, skip)
test_environment.py:83: in my_walk_packages
    yield from my_walk_packages(path, info.name+'.', onerror, skip)
test_environment.py:71: in my_walk_packages
    onerror(info.name)
test_environment.py:68: in my_walk_packages
    __import__(info.name)
</path/to/deployment>/prerelease/apps/base_conda/envs/payu-dev/lib/python3.10/site-packages/backports/zstd/_cffi/__init__.py:1: in <module>
    from backports.zstd._cffi._common import (
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    import enum

>   from backports.zstd._zstd_cffi import ffi as _ffi
E   ModuleNotFoundError: No module named 'backports.zstd._zstd_cffi'

```

</details>


This issue has come up now because `urllib3` (used in `requests`) recently had an update which now uses `backports.zstd`.

There's a couple solutions:
- Pinning libraries for a few weeks to see if issues resolves. From the release notes (https://urllib3.readthedocs.io/en/stable/changelog.html), it looks like there's a couple of security updates in the latest version of `urllib3` so might be worth having that version. I've just tested pinning `backports.zstd` to 1.1.0 (latest is 1.2.0), and it still has the same error.

- Add `backports.zstd._cffi` to list of libraries that allow loading, but pass any exceptions in the tests. So when exceptions no longer triggered, it can be removed: see https://github.com/ACCESS-NRI/model-release-condaenv/pull/46/files which now passes the tests.

--- a/environments/payu-dev/testconfig.yml
+++ b/environments/payu-dev/testconfig.yml
@@ -96,3 +96,4 @@ exception:
 - fontTools.ufoLib #
 - pandas.core._numba.kernels #
 - scipy._lib.array_api_compat.cupy #
+- backports.zstd._cffi #

